includes/options/class-amp-user-manager.php

```diff
@@ -0,0 +1,119 @@
+<?php
+/**
+ * Class AMP_User_Manager.
+ *
+ * @since 1.6.0
+ *
+ * @package AMP
+ */
+
+/**
+ * Class AMP_User_Manager
+ *
+ * @since 1.6.0
+ */
+final class AMP_User_Manager {
```

Comment:
I'm guessing that this should be implemented as a `Service` similar to how we're to be doing things for plugin suppression (#4657) and mobile redirection (#4796). /cc @schlessera

Reply:
Good call. I was thinking it worked like the Gutenberg `editPost` functions in that you have to send back all the meta, but on closer inspection we only have to send the REST endpoint the metadata we're working with. Updated.

Reply:
Yes, I see what's going on here. I have moved the class into the new `src` directory, but it would be good to have @schlessera take a look to make sure I've put everything in the right places.

Reply:
I'm currently adapting the services subsystem to make it more flexible (it was a very bare-bones implementation to get started).
I'll take a look at this as soon as that adaptation is ready.

Reply:
@schlessera While my other updates from Weston's feedback are in progress, I have pushed a change to adapt the new class to the services system. In this case it simply involved (1) moving my new class into the `src` directory, (2) adding it to the `ALL` array in the `Services` class, and (3) changing its `init` method to `register`. I got stuck when looking for a way to access the class instance from elsewhere (was looking for a `Services::get_instance` method or similar) but ended up not needing to do that for this case.

I'll request your review when I have the other updates pushed.